language/move-vm/runtime/src/loader.rs

```diff
@@ -1050,12 +1063,22 @@ impl Function {
             None
         };
         let scope = Scope::Module(module_id);
+        let parameters = module.signature_at(handle.parameters).clone();
         // Native functions do not have a code unit
         let (code, locals) = match &def.code {
-            Some(code) => (code.code.clone(), module.signature_at(code.locals).clone()),
+            Some(code) => (
+                code.code.clone(),
+                Signature(
+                    parameters
+                        .0
+                        .iter()
+                        .cloned()
+                        .chain(module.signature_at(code.locals).0.iter().cloned())
```

Comment:
Same nit as above. `&` instead of `iter()` and   a single `.cloned()` after the chain